tasks/04_pages/scraper.py

```diff
@@ -0,0 +1,83 @@
+import logging
+import requests
+import time
+
+from bs4 import BeautifulSoup as bs
+
+from common import sql, db_config
+
+
+def add_page_row(case_id: int, 
+                 case_number: str,
+                 raw_text: str):
+    
+    # insert relevant info to pages table in the db
+    try:
+        if db_config.db_type == 'sqlite':
+            query = '''INSERT INTO pages (case_id, case_number, raw_text)
+                            VALUES (?, ?, ?)
+                        '''
+        elif db_config.db_type == 'postgresql':
+            query = """INSERT INTO pages (case_id, case_number, raw_text) 
+                            VALUES (%s, %s, %s);
+                    """
+                
+        with sql.db_cnx() as cnx:
+            c = cnx.cursor()
+            c.execute(query, (int(case_id), case_number, raw_text))        
+            
+    except Exception as e:
+        print(f'Error adding page to {db_config.pages} table: {e}')
+    finally:
+        c.close()
+        cnx.close()
+
+
+
+
+def threaded_scraper(case):
+    """
+    scraper doesn't necessarily need to be run in threads
+    """
+        
+    # get some useful variables from the inputs
+    case_id, case_number = case
+    
+    # keep track of time
+    t1 = time.time()    
+
+    # try to scrape a page. 
+    try:
+        case_response = requests.get(f'https://www.nlrb.gov/case/{case_number}')
+        case_response.raise_for_status()
+    
+    # if you get an error, log it
+    except requests.exceptions.HTTPError as e:
+        print(f'Error: {e}, case: {case_number}')
+        logging.warning(f'{case_id}, {case_number}, fetch error: {e}')
+        return
```

Comment:
You may want to remove this if you just want to log errors and move on. If not, then you'll want to raise the exception in addition to logging instead of returning.

Reply:
i think raising the exception makes sense here...